ui/src/views/user/login/LoginView.vue

```diff
@@ -116,27 +121,40 @@ export default {
   },
   methods: {
     onFormSubmit() {
-      userLogin(this.formData).then((response) => {
-        const authKey = response["message"].split("Access token is ")[1];
-        const userData = {
-          accountType: 0,
-          email: this.formData.email
-        };
-        this.$cookies.set("access_token_cookie", authKey);
-        this.$store.dispatch("user/setToken", authKey);
-        this.$store.dispatch("user/setUserData", userData);
-        this.$store.dispatch("user/checkAccountTypeAndSetMenuItems");
+      this.isLoading = true;
+      userLogin(this.formData)
+        .then((response) => {
+          const authKey = response["message"].split("Access token is ")[1];
+          const userData = {
+            accountType: 0,
+            email: this.formData.email
+          };
+          this.$cookies.set("access_token_cookie", authKey, { expires: 7 });
+          this.$store.dispatch("user/setToken", authKey);
+          this.$store.dispatch("user/setUserData", userData);
+          this.$store.dispatch("user/checkAccountTypeAndSetMenuItems");
 
-        userNotActivated().then((userActivationStatus) => {
-          if (userActivationStatus["message"].includes(" is not activated")) {
-            this.$store.dispatch("user/setHasBeenActivated", false);
-            this.$router.push({ name: "user-confirm" }).catch(() => {});
-          } else {
-            this.$store.dispatch("user/setHasBeenActivated", true);
-            this.$router.push({ name: "user-surveys" }).catch(() => {});
-          }
+          userNotActivated()
+            .then((userActivationStatus) => {
+              if (
+                userActivationStatus["message"].includes(" is not activated")
+              ) {
+                this.$store.dispatch("user/setHasBeenActivated", false);
+                this.$router.push({ name: "user-confirm" }).catch(() => {});
+              } else {
+                this.$store.dispatch("user/setHasBeenActivated", true);
+                this.$router.push({ name: "user-surveys" }).catch(() => {});
+              }
+            })
+            .catch(() => {
+              this.$router.push({ name: "user-surveys" }).catch(() => {});
+            });
```

Comment:
So, if there is an error during checking user activation status, it will redirect to user-surveys anyway. I think this is fine, but I personally prefer if the redirect goes to user-confirm. But if this better for making life easier during testing and demo, let's stick with this.

Reply:
I think its the router handler for user-confirm page that should be handled instead; the router should redirect every routes to user-confirm page when hasBeenActivated value is false, but I dont want to touch the code since im still not in my prime condition and routing is prone of bugs XD

Reply:
aha, ok. maybe lets keep it as it is now. you can merge if you want, i will merge my PR tomorrow after re-basing

Reply:
Alright, merged